src/groupeddataframe/grouping.jl

```diff
@@ -1125,3 +1139,20 @@ function DataFrame(gd::GroupedDataFrame)
     resize!(idx, doff - 1)
     parent(gd)[idx, :]
 end
+
+"""
+    groupindices(gd::GroupedDataFrame)
+
+Return a vector of group indices for each row of `parent(gd)`.
+If for some row this function returns an index `i` this means that
+a data frame `gd[i]` contains this row.
+Rows not present in any of `gd` groups have their index set to `missing`.
+"""
+groupindices(gd::GroupedDataFrame) = replace(gd.groups, 0=>missing)
+
+"""
+    groupvars(gd::GroupedDataFrame)
+
+Return a vector of column indices in `parent(gd)` used for grouping.
```

Comment:
Maybe returning names would be nicer and more useful?

Reply:
OK. Fixed. but it will be a bit slower for the user (integer indices are faster to use later).

Reply:
Yeah, that's why I wrote this as a question. I guess we could add an argument to choose the return type if that really matters for performance.

Reply:
I would leave it as is for now as returning names can be the default so the option can be added later if needed. However, I have opened https://github.com/JuliaData/DataFrames.jl/pull/1712 as this conversion is potentially useful also in other cases.